peachjam/views/legislation.py

```diff
@@ -17,3 +23,46 @@ class LegislationListView(FilteredDocumentListView):
 class LegislationDetailView(BaseDocumentDetailView):
     model = Legislation
     template_name = "peachjam/legislation_detail.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["notices"] = self.get_notices()
+        return context
+
+    def get_notices(self):
+        notices = super().get_notices()
+
+        repeal = self.get_repeal_info()
+        if self.object.repealed and repeal:
+            msg = "This {} was repealed on {} by <a href='{}'>{}</a>"
+            notices.append(
+                {
+                    "type": messages.WARNING,
+                    "html": format_html(
+                        msg.format(
+                            self.object.metadata_json["type_name"],
+                            repeal.date,
+                            repeal.repealing_uri,
+                            repeal.repealing_title,
+                        )
+                    ),
+                }
+            )
+
+        # TODO: Out-of-date legislation
+
+        # TODO: Up-to-date legislation
+
+        return notices
+
+    def get_repeal_info(self):
+        repeal_info = json.dumps(self.object.metadata_json.get("repeal", None))
```

Comment:
This looks odd to me -- I thought the data in `metadata_json` should already be decoded into dictionaries, since it's stored as JSONField?